app/models/server/raw/VoipRooms.ts

```diff
@@ -0,0 +1,171 @@
+import { Cursor, FilterQuery, WithoutProjection, FindOneOptions } from 'mongodb';
+
+import { BaseRaw } from './BaseRaw';
+import { IVoipRoom } from '../../../../definition/IRoom';
+import { Logger } from '../../../../server/lib/logger/Logger';
+
+type T = IVoipRoom;
+export class VoipRoomsRaw extends BaseRaw<T> {
+	logger = new Logger('VoipRoomsRaw');
+
+	findOneOpenByVisitorToken(visitorToken: string, options?: undefined | WithoutProjection<FindOneOptions<T>>): Promise<T | null> {
+		const query: FilterQuery<T> = {
+			't': 'v',
+			'open': true,
+			'v.token': visitorToken,
+		};
+		if (options === undefined) {
+			return this.findOne(query);
+		}
+		return this.findOne(query, options);
+	}
+
+	findOpenByVisitorToken(visitorToken: string, options?: undefined | WithoutProjection<FindOneOptions<T>>): Cursor<T> {
+		const query: FilterQuery<T> = {
+			't': 'v',
+			'open': true,
+			'v.token': visitorToken,
+		};
+		if (options === undefined) {
+			return this.find(query);
+		}
+		return this.find(query, options);
+	}
+
+	/**
+	 * (Amol) Delete this comment and code (if needed)
+	 * after the review.
+	 *
+	 * We have findOneByIdOrName. Do we still need findOneVoipRoomById?
+	 *
+	 */
+	async findOneVoipRoomById(id: string, options?: undefined | WithoutProjection<FindOneOptions<T>>): Promise<T | null> {
+		const query: FilterQuery<T> = {
+			t: 'v',
+			_id: id,
+		};
+		if (options === undefined) {
+			return this.findOne(query);
+		}
+		return this.findOne(query, options);
+	}
+
+	findOneByIdOrName(_idOrName: string, options?: undefined | WithoutProjection<FindOneOptions<T>>): Promise<T | null> {
+		const query: FilterQuery<T> = {
+			t: 'v',
+			$or: [
+				{
+					_id: _idOrName,
+				},
+				{
+					name: _idOrName,
+				},
+			],
+		};
+		if (options === undefined) {
+			return this.findOne(query);
+		}
+		return this.findOne(query, options);
+	}
```

Comment:
```suggestion
```

You're not using this anywhere (and it's better not to)

Reply:
Fixed.